epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java

```diff
@@ -603,6 +605,36 @@ private MethodSpec generateSetter(ClassToGenerateInfo helperClass, AttributeInfo
         .build();
   }
 
+  private MethodSpec generateReset(ClassToGenerateInfo helperClass) {
+    Builder builder = MethodSpec.methodBuilder("reset")
+        .addModifiers(Modifier.PUBLIC)
+        .returns(VOID);
+
+    for (AttributeInfo attributeInfo : helperClass.getAttributeInfo()) {
+      if (!attributeInfo.hasFinalModifier()) {
+        String attributeName = attributeInfo.getName();
+        TypeName attributeType = attributeInfo.getType();
+        if (attributeType == BOOLEAN) {
+          builder.addStatement("this.$L = false", attributeName);
+        } else if (attributeType == BYTE || attributeType == SHORT || attributeType == INT) {
+          builder.addStatement("this.$L = 0", attributeName);
+        } else if (attributeType == LONG) {
+          builder.addStatement("this.$L = 0L", attributeName);
+        } else if (attributeType == FLOAT) {
+          builder.addStatement("this.$L = 0.0f", attributeName);
+        } else if (attributeType == DOUBLE) {
+          builder.addStatement("this.$L = 0.0d", attributeName);
+        } else if (attributeType == CHAR) {
+          builder.addStatement("this.$L = '\\u0000'", attributeName);
```

Comment:
I didn't realize that char's default is `'\u0000', cool. From http://stackoverflow.com/questions/9909333/whats-the-default-value-of-char and http://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html it seems this is equivalent to `0`. How about assigning it to 0 instead since that might be the better understood default value, and we could combine it with the byte, short, int block to simplify.